<launch>
	<param name="/use_sim_time" value="true"/>

	<node pkg="stage_ros" type="stageros" name="stageros_node" args="$(find person_follower)/worlds/simple.world" />
	
	<node pkg="person_follower" type="simulation_interface_node" name="simulation_interface_node" output="screen" />

	<node pkg="person_follower" type="person_follower_node" name="person_follower_node" output="screen">
		<remap from="/husky/odom" to="/robot_0/odom" />
		<remap from="/husky/cmd_vel" to="/robot_0/cmd_vel" />
	</node>

	<include file="$(find person_follower)/launch/ekf.launch" />
	<!-- <node pkg="slam_karto" type="slam_karto" name="slam_karto" output="screen"> -->
	<node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
		<remap from="scan" to="robot_0/base_scan_1"/>
		<param name="odom_frame" value="robot_0/odom_combined"/>
		<param name="base_frame" value="robot_0/base_link"/>
		<param name="map_frame" value="map"/>
	    <param name="map_update_interval" value="1"/>
	    <param name="resolution" value="100"/>
	    <param name="xmin" value="-200"/>
	    <param name="ymin" value="-200"/>
	    <param name="xmax" value="200"/>
	    <param name="ymax" value="200"/>
	    <rosparam command="load" file="$(find person_follower)/config_nav/mapper_params.yaml" />
	</node>

	<include file="$(find person_follower)/config_nav/move_base.xml" />

	<node pkg="rviz" type="rviz" name="rviz" output="screen" />
	
	<node pkg="joy" type="joy_node" name="joy_node" output="screen">
		<remap from="/joy" to="/teleop/joy" />
		<param name="dev" type="string" value="/dev/input/js1" />
		<param name="autorepeat_rate" value="10" />
	</node>

	<node pkg="teleop_twist_joy" type="teleop_node" name="teleop_node">
		<remap from="joy" to="teleop/joy" />
		<remap from="cmd_vel" to="robot_1/cmd_vel" />
	</node>
</launch>